docs/references/nextjs/auth-middleware.mdx

```diff
@@ -95,6 +159,42 @@ export const config = {
 };
 ```
 
+You can use the `req` object to match against urls. The following example will make all routes but `/dashboard` public.
+
+```ts filename="middleware.ts" {4}
+import { authMiddleware } from "@clerk/nextjs";
+
+export default authMiddleware({
+  publicRoutes: req => !req.url.includes('/dashboard')
+})
+
+export const config = {
+  matcher: ['/((?!.+\\.[\\w]+$|_next).*)', '/', '/(api|trpc)(.*)'],
+};
+```
+
+```ts filename="middleware.ts"
+export default authMiddleware({
+  publicRoutes: req => !req.url.includes('/dashboard')
+})
+```
+
+
+You can also use regex to match routes. The following is an example of a negative assertion that makes only the `/admin` protected.
```

Comment:
For this example and the one below, do we want the full middleware or just the second example with the authMiddleware only?

Reply:
@rachelnabors and @alexisintech -- any preference with regards to this? Should I got with the longer, full (first) code example or the shorter (second) one only showing the `authMiddleware` code.

Reply:
oh I think the longer !! it should be the full code example, and then highlight the particular lines that represent what the copy talked about (as you have done, which is perfect).